### DISCUSSION TITLE
Can this be done with STT payload?

### DISCUSSION THREAD

**Discussion by jrohlandt** [deepgram/community#806] [General help] [answered]
Sorry for the bad title. 

What we are doing is using the speech to text api to get the transcription of a users voice over. We are mainly interested in the words and timestamps.

However in some cases our user wants to provide their own text. 
So instead of using the text generated by Deepgram. We use the users text and we use the timestamps from the Deepgram payload.

Problem is that the user text is sometimes quite a bit different. E.g. they might decide to use numerals like 24 instead of "twenty four". They might prefer to type bootcamp instead of boot camp. 
They might type their brand name as "SuperCarRepairs" but Deepgram might transcribe it as "super" "car" "repairs".
And there are many different cases.

So I was wondering is there a way that Deepgram could take the users transcript (we upload it along with the audio file) and update the payload accordingly while still keeping the timestamps correct?

I asked ChatGPT to do it and it can do it, so we might try something with OpenAI's api but I want to check does Deepgram offer such a service already?

User transcript:
`Hi this is Bill from SuperCarRepairs and this week we will change the suspension on a 87 Mustang.`

Deepgram payload:
`{
"words": [{Hi, ...}, {this, ...}, {is, ...}, {Bill, ...}, {from, ...}, {super, 0.223, 0.257}, {car, 0.257, 0.310}, {repairs, 0.310, 0.389}, ...]
}`

Expected payload:
`{
"words": [{Hi, ...}, {this, ...}, {is, ...}, {Bill, ...}, {from, ...}, {supercarrepairs, 0.223, 0.389}, ...]
}`

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
Thanks for your question, @jrohlandt ,

Unfortunately this is not a feature that we currently offer, but it's an interesting idea! But for now you would have to do this yourself programmatically. I'm not sure which programming language you are using, but something like the `difflib` library in Python could help. Here are the steps you could take (although I'm guessing ChatGPT might have given you similar advice):

1. Normalize Text: Convert the text to lowercase and remove punctuation for better alignment.
2. Align Transcript: Uses difflib.SequenceMatcher to align the words from the user's transcript with the Deepgram payload.
3. Adjust Timestamps: Combine the timestamps from Deepgram's words for combined words in the user transcript.

**Reply by jrohlandt** [reply]
Perfect solution thanks. Ended up using https://github.com/qiao/difflib.js for js.
